### PR TITLE
Set up continuous integration with sge

### PR DESCRIPTION
**Pull request by arjoly**

### CODE REVIEW

**Comment by arjoly** [comment]
It will fix issue #18

**Comment by arjoly** [comment]
@ogrisel This should be good to go. Do you want to have a look before I merge this?

File: clusterlib/tests/test_scheduler.py [hunk 1]
```diff
@@ -61,6 +61,56 @@ def test_fixed_backend():
...
+
+    user = getuser()
+    job_name = "test-sleepy-job"
+
+    # Launch a sleepy slurm job
+    sleep_job = submit(job_command="sleep 600", job_name=job_name,
+                       backend="sge", time="700", memory=100)
+    os.system(sleep_job)
+
+   # Get job id
```
Comment (ogrisel) [inline clusterlib/tests/test_scheduler.py]: indent issue

File: clusterlib/tests/test_scheduler.py [hunk 2]
```diff
@@ -61,6 +61,56 @@ def test_fixed_backend():
...
+    job_id = None
+    try:
+        out = subprocess.check_output("qstat -xml -u {0}".format(user),
```
Comment (ogrisel) [inline clusterlib/tests/test_scheduler.py]: I think you can do `"qstat -xml -u {0} -j {1}".format(user, job_name)` directly.

Comment (ogrisel) [inline clusterlib/tests/test_scheduler.py]: BTW, it's better to use the `check_output(["qstat", "-xml", "-u", user, "-j", job_name])` to avoid shell injection security holes.

**Comment by ogrisel** [comment]
Apart from my comment, it looks great.

File: clusterlib/tests/test_scheduler.py (hunk 2 above)
Comment (arjoly) [inline clusterlib/tests/test_scheduler.py]: > I think you can do "qstat -xml -u {0} -j {1}".format(user, job_name) directly.

Cool, I have never realised that it was possible to query `qstat` with `-j` using the job name!

**Comment by arjoly** [comment]
@ogrisel I hope that the injection is fixed now.

File: clusterlib/scheduler.py [hunk 3]
```diff
@@ -78,32 +78,32 @@ def _get_backend(backend="auto"):
...
-        command += " -u {}".format(user)
+        command.extend(["-u", user])
 
     try:
         with open(os.devnull, 'w') as shutup:
-            xml = subprocess.check_output(command, shell=True, stderr=shutup)
+            xml = subprocess.check_output(command, stderr=shutup)
             tree = ElementTree.fromstring(xml)
             return [leaf.text for leaf in tree.iter("JB_name")]
-    except subprocess.CalledProcessError:
```
Comment (ogrisel) [inline clusterlib/scheduler.py]: Why this change?

File: clusterlib/tests/test_scheduler.py [hunk 4]
```diff
@@ -105,7 +161,8 @@ def test_queued_or_running_jobs_slurm():
         assert_in(job_name, running_jobs)
     finally:
         # Make sure to clean up even if there is a failure
-        os.system("scancel %s" % job_id)
+        if job_id is not None:
+            os.system("scancel %s" % job_id)
```
Comment (ogrisel) [inline clusterlib/tests/test_scheduler.py]: `job_id` is presumably not injectable by the user by a `subprocess.call(['scancel', job_id], shell=True)` would also be safer here.

**Comment by ogrisel** [comment]
LGTM! I did not review the SGE conf itself as I don't know how it works but I trust travis ;)

File: clusterlib/scheduler.py (hunk 3 above)
Comment (arjoly) [inline clusterlib/scheduler.py]: Apparently, subprocess is checking that thé program existe. Thus i have an oserror here.

File: clusterlib/tests/test_scheduler.py (hunk 4 above)
Comment (arjoly) [inline clusterlib/tests/test_scheduler.py]: Thanks, i will fix those!

File: clusterlib/scheduler.py (hunk 3 above)
Comment (ogrisel) [inline clusterlib/scheduler.py]: alright

**Comment by arjoly** [comment]
Thanks @ogrisel, it's merged!
